experimental/javascript.html

```diff
@@ -0,0 +1,161 @@
+<script>
+
+  function getSelectedRange(rangePicker,idName){
+    rangePicker.innerHTML = "Selecting"; // Change the button value while getting range
+    rangePicker.disabled = true;         // Disable the button while getting range
+
+    var RangeObj = {
+      "rangePicker" : rangePicker,
+      "idName" : idName
+    }
+    google.script.run                           // Executes a Apps Script function - getSelectedRange
+          .withSuccessHandler(updateRangeField) // updateRangeField will be called upon successfull completion of Apps Script function
+          .withUserObject(RangeObj)             // To pass argument to updateRangeField
+          .getSelectedRange();                  
+    return;
+  }
+        
+  // Function to be called on success
+  function updateRangeField(range,RangeObj){
+    document.getElementById(RangeObj.idName).value = range;      // Update the text field value
+    RangeObj.rangePicker.innerHTML = "Get Selected Range";       // Reset the button value
+    RangeObj.rangePicker.disabled = false;                       // Enable the button
+  }
+      
+  var ids=0;  //counter for slices created
+  var idg=0;  //counter for groups created
+  var idd=0;  //counter for dimensions created
+    
+  function funcAddSlice(){
+    ids++;   //increment number of slices created
+    var sliceId='s'+ids;  //id of the slice element that will be created
+    var html= '<div name="filter" class="box-slice">'+
+                '<button name="delete-filter" class="red btn-del-filter" onclick="removeElement(\''+sliceId+'\')">x</button>'+
+                '<br>'+
+                '<div id="part1-slice" class="block">'+
+                  '<label for="slice-col'+ids+'" class="inline">Slice Column Name:&ensp;</label>'+
+                  '<input type="text" name="slice-col" id="slice-col'+ids+'" class="width90" placeholder="column">'+
+                '</div>'+
+                '<div id="part2-slice" class="block">'+
+                  '<label for="slice-op'+ids+'" class="inline">Filter Operation:&ensp;</label>'+
+                  '<select id="slice-op'+ids+'" name="slice-op" class="width100">'+
+                    '<option>Equal to</option>'+
```

Comment:
can we have a helper function that does the same thing. can just call it everywhere you generate options.

Reply:
Yeah, I will write a function for it